python/ray/tests/test_reconstruction.py

```diff
@@ -10,7 +10,12 @@
     wait_for_condition, )
 
 
-def test_cached_object(ray_start_cluster):
+def shutdown_upon_termination(cluster):
```

Comment:
You should make this a fixture, else it won't be run when . You can just do something like:
```
cluster = Cluster()
yield cluster
ray.shutdown()
cluster.shutdown()
```

Reply:
Addressed